hasher-matcher-actioner/hmalib/matchers/matchers_base.py

```diff
@@ -260,13 +304,19 @@ def get_metadata_objects_from_match(
     def get_index(self, signal_type: t.Type[SignalType]) -> SignalTypeIndex:
         # If cached, return an index instance for the signal_type. If not, build
         # one, cache and return.
-        if not signal_type in self._cached_indexes:
-            index_cls = INDEX_MAPPING[signal_type]
+        max_custom_threshold = (
```

Comment:
super-nit: weird use of paranthesis. Easily confusable as a tuple. Did black do this?

Reply:
We end up computing max_custom_threshold and index_cls even if cached. Can avoid.

Reply:
re: parenthesis -> black did do this.

We should check because if it changed it is possible the cached index could be the incorrect type. This is pretty niche but here is where using cached index blindly breaks down.

1. No custom threshold or custom threshold < 31 is set.
2. Matcher lambda is invoked and because of current config state PDQIndex is cached for pdq signal type
3. Custom threshold is set for > 31 (matcher that should be used is now PDQFlatIndex)
4. Matcher lambda is invoked again where `PG_CONFIG_CACHE_TIME_SECONDS` has a set value that is pass but `_cached_indexes` is still in the lambda/matchers memory.
5. without this check: `get_index` would incorrectly return `PDQIndex` because it didn’t recompute (or at least check the other cache of) `max_custom_threshold`